src/annotation_tool/project.py

```diff
@@ -77,9 +78,9 @@ def annotate(self, index, text):
         """Changes the text of the given annotation."""
         annotation = self.annotations[index]
         if not annotation.modified:
-            self.modified_annotations[annotation.file] = True
+            self.modified_annotations.append(annotation.file)
         if not text:
-            self.modified_annotations[annotation.file] = False
+            self.modified_annotations.remove(annotation.file)
```

Comment:
why is removing a text not a modification?

Reply:
This is so that removing the text marks it as "Not finished". Maybe there should be two colors, orange for modified but empty, green for modified and filled in.

Reply:
Actually, I would assume, that removing text is a modification. Maybe something similar to "Mark as unread" might be helpful as an alternative.